### PR TITLE
Refactor FXIOS-9719 Update Twitter name String to X

### PR DESCRIPTION
**Pull request by ih-codes**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9719)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21362)

## :bulb: Description
Update the String for "Twitter" to be "X" (new name).

Screenshot 2024-08-26 at 4 25 52 PM

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Strings.swift [hunk 1]
```diff
@@ -6534,6 +6534,11 @@ extension String {
         tableName: nil,
         value: nil,
         comment: "Tile title for Twitter")
+    public static let DefaultSuggestedX = MZLocalizedString(
+        key: "SuggestedSites.X",
```
Comment (ih-codes) [inline firefox-ios/Client/Frontend/Strings.swift:6538]: "X" seemed a bit vague for a key, but I can change it to that if we want to match the others?

File: firefox-ios/Client/Frontend/Strings.swift [hunk 2]
```diff
@@ -6534,6 +6534,11 @@ extension String {
...
+        tableName: nil,
+        value: "X",
+        comment: "Tile title for X (formerly Twitter)")
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Strings.swift:6541]:
```suggestion
    public static let DefaultSuggestedX = MZLocalizedString(
        key: "SuggestedSites.X.Title.v131",
        tableName: "SuggestedSites",
        value: "X",
        comment: "Title for X (formerly Twitter) tile in the suggested sites section of the homepage.")
```

**Review by adudenamedruby — APPROVED** [review]

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ActivityStreamTest.swift [hunk 3]
```diff
@@ -10,7 +10,7 @@ let newTopSite = [
     "topSiteLabel": "Mozilla",
     "bookmarkLabel": "Internet for people, not profit — Mozilla (US)"
 ]
-let allDefaultTopSites = ["Facebook", "YouTube", "Amazon", "Wikipedia", "Twitter"]
+let allDefaultTopSites = ["Facebook", "YouTube", "Amazon", "Wikipedia", "X"]
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ActivityStreamTest.swift:13]: Have you found all occurrences of "Twitter"? There may be more.

Comment (ih-codes) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/ActivityStreamTest.swift:13]: I believe so--I did a quick search and the tests are now passing.

Comment (ih-codes) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/ActivityStreamTest.swift:13]: There was a use of "Twitter" in `oldHistoryEntries` and some commented out tests. But those did not appear to be causing failures. 🤔

**Review by adudenamedruby — APPROVED** [review]

**Pull request merged**